src/librbd/operation/ResizeRequest.cc

```diff
@@ -212,12 +212,18 @@ Context *ResizeRequest<I>::send_grow_object_map() {
   }
   image_ctx.aio_work_queue->unblock_writes();
 
+  CephContext *cct = image_ctx.cct;
   if (m_original_size == m_new_size) {
     if (!m_disable_journal) {
       this->commit_op_event(0);
     }
     return this->create_context_finisher();
   } else if (m_new_size < m_original_size) {
+    if (!m_allow_shrink) {
```

Comment:
This check needs to be performed before the journal event is recorded within the ```send_append_op_event``` method. Once the journal event is recorded, bailing out is considered a split-brain event.